### PR TITLE
Separate the Windows and Linux binaries

### PR DESCRIPTION
**Pull request by simonmarty**
### Why is this change being made?

1. Separate the Windows and linux binaries into separate paths and buckets for signing.

### What is changing?

1. Simple if/else statements to populate the S3 bucket name and the object path from GH secrets.

### Related Links
- **Issue #, if available**:

---

### How was this tested?

1. 

### When testing locally, provide testing artifact(s):

1. 

---

## Reviewee Checklist

**Update the checklist after submitting the PR**

- [x] I have reviewed, tested and understand all changes
  *If not, why:*
- [x] I have filled out the Description and Testing sections above
  *If not, why:*
- [x] Build and Unit tests are passing
  *If not, why:*
- [x] Unit test coverage check is passing
  *If not, why:*
- [x] Integration tests pass locally
  *If not, why:*
- [ ] I have updated integration tests (if needed)
  *If not, why:*
- [x] I have ensured no sensitive information is leaking (i.e., no logging of sensitive fields, or otherwise)
  *If not, why:*
- [ ] I have added explanatory comments for complex logic, new classes/methods and new tests
  *If not, why:*
- [ ] I have updated README/documentation (if needed)
  *If not, why:*
- [ ] I have clearly called out breaking changes (if any)
  *If not, why:*

---

## Reviewer Checklist

**All reviewers please ensure the following are true before reviewing:**

- Reviewee checklist has been accurately filled out
- Code changes align with stated purpose in description
- Test coverage adequately validates the changes

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by i-am-SR** [comment]
Claude has good suggestions worth fixing including `Potential silent misrouting`, `Leading slash in S3 key when the bin-path secret is empty` and `Inconsistent approach between the two files`.

The testing section is empty in the description - are we planning to test after merging?

**Comment by simonmarty** [comment]
I'm going to fix the naming inconsistency for the secrets. The other edge cases strike me as overly defensive. I think it's obvious that all secrets can be expected to be set but I can add a step that actually validates that at the top. I think that's cleaner than having a check in every place where the secrets are used.

**Comment by simonmarty** [comment]
I'm testing this on my fork. I will post results shortly.

File: .github/workflows/staging.yml [hunk 1]
```diff
@@ -64,5 +83,10 @@ jobs:
           aws-region: us-west-2
 
       - name: Upload artifacts to S3
+        env:
+          BUCKET: ${{ runner.os == 'Windows' && secrets.STAGING_WINDOWS_BUCKET || secrets.STAGING_BUCKET }}
```
Comment (i-am-SR) [inline .github/workflows/staging.yml:87]: we can create a backlog for making this logic consistent across the 2 workflows for the sake of reducing cognitive overhead for future maintainers.

**Review by i-am-SR — APPROVED** [review]
Approved assuming successful testing on fork.

**Review by reyhankoyun — APPROVED** [review]
